### PR TITLE
feature(table-data-source): add the stream searched

### PR DESCRIPTION
**Pull request by niwolf**
@plentymarkets/team-terra

#### Must be done by Terra

Testing

-   [ ] Person 1 (mandatory)

    Browser support (relevant for changes in scss / appearance of components)

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

-   [ ] Person 2 (mandatory)

    Browser support

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

Information transfer

-   [ ] Inform Ceres about changes in `TerraFormComponent` (optional: relevant for changes in this component)

Terra Basic Plugin

-   [ ] Adapt changes (optional: relevant for version updates, global design changes, l10n)

---

#### Must be done by every developer

Please inform a member of Terra (@plentymarkets/team-terra) to get the upper part of the checklist done (with urgency or deadline).

Documentation

-   [ ] Changelog (optional: relevant for every change which influences the API)
-   [ ] Example (updated or created)
-   [ ] JSDoc (optional: relevant for every change which influences the API)
-   [ ] [Google spreadsheet](https://docs.google.com/spreadsheets/d/1OINnux8TEoitV-qdAxqUQaf9oGI_fqFVwfRWenRFfhI/edit#gid=0) (relevant when deprecating public APIs or features)

Testing

-   [ ] Unit Test (updated or created)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# 11.4.0 (XX.06.2021)
+
+### Features
+
+-   **terra-table-data-source** now has a stream that emits whenever the data source is requested.
```
Comment (pweyrich) [inline CHANGELOG.md:5]:
```suggestion
-   **table** introduced `searched$` property to `TerraTableDataSource` which notifies whenever data has been requested from the server.
```

File: src/lib/components/table/terra-table-data-source.ts [hunk 2]
```diff
@@ -102,6 +102,11 @@ export abstract class TerraTableDataSource<D, F = unknown> extends DataSource<D>
     }
     private _paginator: MatPaginator | undefined;
 
+    /** A stream that emits whenever the data source is requested. */
```
Comment (pweyrich) [inline src/lib/components/table/terra-table-data-source.ts:105]:
```suggestion
    /** A stream that emits whenever data has been requested from the server. */
```

File: src/lib/components/table/terra-table-data-source.ts [hunk 3]
```diff
@@ -102,6 +102,11 @@ export abstract class TerraTableDataSource<D, F = unknown> extends DataSource<D>
     }
     private _paginator: MatPaginator | undefined;
 
+    /** A stream that emits whenever data has been requested from the server. */
+    // tslint:disable-next-line:member-ordering
+    public searched$: Observable<boolean>;
+    private _searched$: BehaviorSubject<boolean> = new BehaviorSubject(false);
```
Comment (pweyrich) [inline src/lib/components/table/terra-table-data-source.ts:108]: Nur ein Vorschlag zu einem alternativen Namen:
```suggestion
    public dataRequested$: Observable<boolean>;
    private _dataRequested$: BehaviorSubject<boolean> = new BehaviorSubject(false);
```

**Review by pweyrich — DISMISSED** [review]
tested 👍

File: src/lib/components/table/terra-table-data-source.ts (hunk 3 above)
Comment (nappynapster) [reply src/lib/components/table/terra-table-data-source.ts:108]: oder vllt auch dataReceived

**Review by nappynapster — DISMISSED** [review]

**Review by nappynapster — APPROVED** [review]

**Review by pweyrich — APPROVED** [review]

**Pull request merged**